src/main/java/com/thealgorithms/backtracking/ArrayCombination.java

```diff
@@ -10,23 +10,35 @@
 public final class ArrayCombination {
     private ArrayCombination() {
     }
-    private static int length;
 
     /**
-     * Find all combinations of 1..n by creating an array and using backtracking in Combination.java
-     * @param n max value of the array.
-     * @param k length of combination
-     * @return a list of all combinations of length k. If k == 0, return null.
+     * Find all combinations of 1..n using backtracking.
+     *
+     * @param n Max value of the elements.
+     * @param k Length of the combination.
+     * @return A list of all combinations of length k.
+     *         Returns an empty list if k is 0 or n is less than k.
      */
-    public static List<TreeSet<Integer>> combination(int n, int k) {
-        if (n <= 0) {
-            return null;
+    public static List<List<Integer>> combination(int n, int k) {
+        if (k <= 0 || n < k) {
+            return new ArrayList<>(); // Return empty list for invalid input
         }
```

Comment:
There is nothing bad with `n >= 0` and `k == 0`. Please throw `IllegalArgumentException` if one of the inputs is negative or `n < k`.

Please add tests checking if the exception is thrown.